webapp/src/pages/ComponentsBook/ComponentsExamples/HeaderExample.tsx

```diff
@@ -14,7 +15,7 @@ export const HeaderExample = () => (
       {placeholderTitle}
     </Header>
     <Header
-      navigationFigure={randomAvatar}
+      navigationFigure={JSON.stringify(avProps)}
```

Comment:
This is questionable, but it appeared to me as the simplest way to achieve the passing of the coordinates to the avatar, here, with the least amount of changes to the component as it is.

Reply:
question: What do we think about splitting this into two props: `iconVariant` and `avatarProps`? And require one or the other.
Or pass the JSX element directly?

I'm concerned that we add hacks on top of hacks (the previous way was also a hack :sweat_smile:) and risk creating something pretty hard to understand down the road.

Reply:
I think it's a good idea. Will do.

Reply:
How would you do the part where we already have two union types for link and button? If we add two variants, I'll have to add two variants for button and two variants for link. Isn't this making it very complicated very soon? (There is a technical term for this problem, but I cannot recall it)

Reply:
We could do something like this:

<details>

```diff
diff --git a/webapp/src/shared/components/Header/Header.tsx b/webapp/src/shared/components/Header/Header.tsx
index 848bdd6..012b056 100644
--- a/webapp/src/shared/components/Header/Header.tsx
+++ b/webapp/src/shared/components/Header/Header.tsx
@@ -8,25 +8,37 @@ import './Header.css';
 import React from 'react';
 
 type HeaderCommon = {
-  navigationFigure: IconVariant | string;
   statusVariant?: StatusVariant;
   children: string;
 };
 
-type HeaderLinkProps = HeaderCommon & {
+type HeaderLinkNavigation = {
   navigationUrl: string;
   onClick?: never;
 };
 
-type HeaderButtonProps = HeaderCommon & {
+type HeaderButtonNavigation = {
   navigationUrl?: never;
   onClick: React.MouseEventHandler<HTMLButtonElement>;
 };
 
-type HeaderProps = HeaderLinkProps | HeaderButtonProps;
+type HeaderIconFigure = {
+  iconVariant: IconVariant;
+  avatarProps?: never;
+};
+
+type HeaderAvatarFigure = {
+  iconVariant?: never;
+  avatarProps: AvatarProps;
+};
+
+type HeaderProps = HeaderCommon &
+  (HeaderLinkNavigation | HeaderButtonNavigation) &
+  (HeaderIconFigure | HeaderAvatarFigure);
 
 export default function Header({
-  navigationFigure,
+  iconVariant,
+  avatarProps,
   navigationUrl,
   onClick,
   statusVariant,
@@ -34,15 +46,10 @@ export default function Header({
 }: HeaderProps) {
   const lNavFigure = (
     <>
-      {navigationFigure in IconVariant ? (
-        <Icon
-          variant={navigationFigure as IconVariant}
-          size={IconSize.SMALL}
-          color={Color.LIGHT}
-        />
-      ) : (
-        <SmallAvatar {...(JSON.parse(navigationFigure) as AvatarProps)} />
+      {iconVariant && (
+        <Icon variant={iconVariant} size={IconSize.SMALL} color={Color.LIGHT} />
       )}
+      {avatarProps && <SmallAvatar {...avatarProps} />}
     </>
   );
   return (

```

</details>


You are right. It is a bit convoluted if we want to keep the types and conditional props.

Reply:
This should be fixed now (following your last proposition). Let me know if this works for you.

Reply:
Thank you! Looks very good!